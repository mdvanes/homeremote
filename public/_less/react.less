// for /react test

// Toggle
.btn-icon {
    font-size: 350%;
}

// Button-Set
.btn-group-justified {
    > a,
    > span {
        font-size: 230%;
    }
}

.margin-top {
    margin-top: 1em;
}

.log {
    color: #fff;
    height: 300px;
}