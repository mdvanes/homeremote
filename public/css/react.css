.btn-icon {
  font-size: 350%;
}
.btn-group-justified > a,
.btn-group-justified > span {
  font-size: 230%;
}
.margin-top {
  margin-top: 1em;
}
.log {
  color: #fff;
  height: 300px;
}
