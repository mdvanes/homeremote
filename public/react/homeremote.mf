CACHE MANIFEST
index.html
../library/bootstrap/css/bootstrap.min.css
../library/bootstrap/fonts/glyphicons-halflings-regular.woff2
../library/bootstrap/fonts/glyphicons-halflings-regular.woff
../library/bootstrap/fonts/glyphicons-halflings-regular.ttf
../css/react.css
build/bundle.js

NETWORK:
*
