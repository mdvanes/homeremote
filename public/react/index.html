<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>React</title>
    
    
  </head>
  <body>
    
    <script src="build/bundle.js"></script>
    
  </body>
</html>
