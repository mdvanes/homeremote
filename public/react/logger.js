/**
 * Created by m.van.es on 6-2-2016.
 */

let logger = {};

let writeLog = (...messages) => {
    let time = new Date();
    let timestamp = time.getHours() + ':' +
        time.getMinutes() + '.' +
        time.getSeconds();
    document.querySelector('.log').value = timestamp + ' ' +
        messages.join(' ') + '\n' +
        document.querySelector('.log').value;
};

logger.log = message => {
    writeLog('INFO:    ', message);
};

logger.error = message => {
    writeLog('ERROR:', message);
};

export default logger;