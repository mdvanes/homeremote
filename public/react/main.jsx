// from https://github.com/code0wl/react-example-es2015/blob/master/app/main.jsx#L5
//import './stylesheets/main.css';
import React from 'react';
import {render} from 'react-dom';
////import FilterableProductTable from './src/filterable-product-table';
import ButtonGroup from './components/button-group';
import Toggle from './components/toggle';
import Log from './components/log';

// init shell
renderShell();

// TODO toggle as single button with icon (instead of 2 buttons)

function renderShell() {
    var shell = document.createElement('div');
    shell.className = 'app-shell container';
    document.body.appendChild(shell);
    render(
        <div>
            <div className="row">
                <div className="col-xs-12">
                    <ButtonGroup label=" 1" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 2" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 3" icon="lamp" id="clickstub"/>
                    <ButtonGroup label=" 4" icon="lamp" id="clickstub"/>
                </div>
            </div>
            <div className="row margin-top">
                {/*<Toggle label=" 2" id="togglestub" icon="lamp"/>
                <Toggle label=" 3" id="togglestub" icon="lamp"/>
                <Toggle label="Switch 4" id="togglestub"/>*/}
                <Toggle label="" id="togglestub" icon="volume-up"/>{/* Broadcast */}
                <Toggle label="" id="togglestub" icon="music"/>{/* Music on server */}
            </div>
            <div className="row margin-top">
                <div className="col-xs-12">
                    <Log/>
                </div>
            </div>
        </div>, shell);
}

