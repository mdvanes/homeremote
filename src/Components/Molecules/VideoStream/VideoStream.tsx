import React, { FC, useEffect, useState } from "react";
import { Paper } from "@material-ui/core";

const VideoStream: FC = () => {
    const [embedSrc, setEmbedSrc] = useState("");
    useEffect(() => {
        fetch(`${process.env.REACT_APP_BASE_URL}/api/nowplaying/radio2embed`)
            .then((url) => url.text())
            .then((url: string) => setEmbedSrc(url));
    }, []);

    return (
        <Paper style={{ aspectRatio: "16/9", overflow: "clip" }}>
            {/* TODO also see https://github.com/streamlink/streamlink/pull/1084 */}
            <iframe
                id="videostream"
                src={embedSrc}
                allowFullScreen={true}
                allow="encrypted-media; autoplay; fullscreen"
                frameBorder="0"
                title="embedded_video_stream"
                style={{
                    width: "calc(100% + 16px)",
                    height: "100%",
                    marginBottom: "-5px",
                }}
            ></iframe>
        </Paper>
    );
};

export default VideoStream;
