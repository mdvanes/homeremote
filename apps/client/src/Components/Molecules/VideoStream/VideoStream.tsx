import { LinearProgress, Paper, Typography } from "@mui/material";
import { FC, useEffect, useState } from "react";

const URL = `${process.env.NX_BASE_URL}/api/video-stream`;

const VideoStream: FC = () => {
    const [isLoading, setIsLoading] = useState(true);
    const [hasError, setHasError] = useState(false);
    const [hash, setHash] = useState("");

    useEffect(() => {
        (async () => {
            setIsLoading(true);
            try {
                const response = await fetch(`${URL}/hash`);
                if (response.status !== 200) {
                    setHasError(true);
                }
                const data = await response.json();
                setHash(data["hash"]);
            } catch (err) {
                setHasError(true);
            }
            setIsLoading(false);
        })();
    }, []);

    if (isLoading) {
        return (
            <Typography variant="body1" textAlign="center">
                VideoStream is loading
                <LinearProgress color="primary" variant="indeterminate" />
            </Typography>
        );
    }

    if (hasError) {
        return (
            <Typography variant="body1" textAlign="center">
                VideoStream failed to load
            </Typography>
        );
    }

    return (
        <Paper style={{ aspectRatio: "16/9", overflow: "clip" }}>
            <video width="100%" controls src={`${URL}?hash=${hash}`} />
        </Paper>
    );
};

export default VideoStream;
