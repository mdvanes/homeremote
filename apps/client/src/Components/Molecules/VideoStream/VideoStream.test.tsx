import React from "react";
import { render, screen } from "@testing-library/react";
import VideoStream from "./VideoStream";
import fetchMock, { enableFetchMocks } from "jest-fetch-mock";

enableFetchMocks();

describe("VideoStream", () => {
    beforeEach(() => {
        fetchMock.resetMocks();
        fetchMock.mockResponse("http://someurl");
    });

    it("renders with stream url", async () => {
        render(<VideoStream />);
        const msg = await screen.findByText("VideoStream failed to load");
        expect(msg).toBeVisible();
    });
});
