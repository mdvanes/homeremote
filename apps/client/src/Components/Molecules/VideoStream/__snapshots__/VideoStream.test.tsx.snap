// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoStream renders with stream url 1`] = `
<body>
  <div>
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiPaper-rounded"
      style="overflow: clip;"
    >
      <iframe
        allow="encrypted-media; autoplay; fullscreen"
        allowfullscreen=""
        frameborder="0"
        id="videostream"
        src="http://someurl"
        style="width: calc(100% + 16px); height: 100%; margin-bottom: -5px;"
        title="embedded_video_stream"
      />
    </div>
  </div>
</body>
`;
